pkg/tcpip/transport/tcp/tcp_test.go

```diff
@@ -6561,3 +6561,71 @@ func TestKeepaliveWithUserTimeout(t *testing.T) {
 		t.Errorf("got c.Stack().Stats().TCP.EstablishedTimedout = %v, want = %v", got, want)
 	}
 }
+
+func TestIncreaseWindowOnReceive(t *testing.T) {
+	// This test ensures that the endpoint sends an ack,
+	// after recv() when the window grows to more than 1 MSS.
```

Comment:
Given that the original bug was when window scaling was disabled we should probably have a subtest or a separate test that tests the correct behaviour when window scaling is disabled.

Another edge case that I feel we need to verify is what happens if the user sets the rcvBuf to a value far lower than MSS. eg

Set the rcvBuf to like 4KB when MSS is 65kb.

Reply:
I removed window scaling from the test, it's really irrelevant.

If user sets recv buffer < MSS, it's her problem. This code won't kick in. This code only makes sure that if buffer (avail) was small, and we did recv() and it's now large, we should immediately ack it, to allow sender to uncork itself (and help senders side silly-window prevention).

I added a code and test to make sure that _increasing_ the recv buffer can indeed generate ack as it should if we suddenly have > MSS avail.